### ISSUE TITLE
Plugs should use `private` instead of `assigns` for storing state

### ISSUE THREAD

**Issue by GregMefford** [spandex-project/spandex#81]
See the Plug docs for details, but libraries aren't supposed to pollute the `assigns` on the Conn.

https://github.com/elixir-plug/plug/blob/master/lib/plug/conn.ex#L69
https://github.com/elixir-plug/plug/blob/master/lib/plug/conn.ex#L84-L89

**Comment by zachdaniel** [comment]
:+1:

**Comment by zachdaniel** [comment]
Thinking about it, we should probably move the plugs stuff over to spandex_phoenix or something. I know plugs are their own thing, but it probably fits more there than it does here.

**Comment by zachdaniel** [comment]
I've updated this issue to represent more modern times. In our 3.0.0 release we should simply remove the plugs from core.

**Comment by GregMefford** [comment]
Agreed, which means we should go ahead and deprecate them now, in favor of the `spandex_phoenix` integration.

**Comment by zachdaniel** [comment]
Yeah, great plan. How do you deprecate a plug? Do we just deprecate the `call` or `init` function?
